pkg/controllers/metrics/pod/controller.go

```diff
@@ -102,28 +103,23 @@ func labelNames() []string {
 	}
 }
 
-// NewController constructs a controller instance
-func NewController(kubeClient client.Client) *Controller {
-	return &Controller{
+// NewController constructs a podController instance
+func NewController(kubeClient client.Client) corecontroller.Controller {
```

Comment:
Is it possible to collapse this with the register method? You should be able to get the client off of the manager

func Register(m *manager.Manager) ... {
  ...
}

Reply:
It is, but then it makes testing a pain we have to inject or set the `kubeClient` in the controller after `New`ing it. Since it's a dependency, I'm more in favor of injecting it than creating a roundabout way to instantiate it.